pkg/apis/core/validation/shoot_test.go

```diff
@@ -4901,6 +4901,37 @@ var _ = Describe("Shoot Validation Tests", func() {
 			})
 		})
 
+		DescribeTable("SeccompDefault",
+			func(version string, SeccompDefaultEnabled bool, SeccompDefaultFeatureGate *bool, matcher gomegatypes.GomegaMatcher) {
+				kubeletConfig := core.KubeletConfig{
+					SeccompDefault: &SeccompDefaultEnabled,
+				}
+				if SeccompDefaultFeatureGate != nil {
+					kubeletConfig.FeatureGates = make(map[string]bool)
+					kubeletConfig.FeatureGates["SeccompDefault"] = *SeccompDefaultFeatureGate
+				}
+
+				errList := ValidateKubeletConfig(kubeletConfig, version, true, nil)
+
+				Expect(errList).To(matcher)
+			},
+
+			Entry("valid configuration", "1.25", true, nil, HaveLen(0)),
+			Entry("valid configuration with set feature gate", "1.25", true, pointer.Bool(true), HaveLen(0)),
+			Entry("do not allow to set SeccompDefault to true when k8s version < 1.25", "1.24", true, nil, ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{
+				"Type":  Equal(field.ErrorTypeForbidden),
+				"Field": Equal("seccompDefault"),
```

Comment:
You could also assert the `Detail` field of the error to have more precise assertions (same below)

Reply:
Thanks!